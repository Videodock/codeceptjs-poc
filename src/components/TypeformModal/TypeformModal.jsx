import React from 'react';
import * as PropTypes from 'prop-types';
import { ReactTypeformEmbed } from 'react-typeform-embed';

export default function TypeformModal ({ url, formRef }) {
  return (
    <ReactTypeformEmbed
      autoOpen={false}
      url={url}
      ref={tf => {
        formRef.current = tf;
      }}
      popup
      hideHeaders
      hideFooter
    />
  );
}

TypeformModal.propTypes = {
  url    : PropTypes.string.isRequired,
  formRef: PropTypes.object.isRequired,
};

