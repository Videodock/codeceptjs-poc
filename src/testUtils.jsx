import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
