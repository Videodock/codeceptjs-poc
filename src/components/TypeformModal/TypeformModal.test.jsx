import React from 'react';
import { render } from '../../testUtils';
import TypeformModal from './TypeformModal';

describe('<TypeformModal />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<TypeformModal url="https://" formRef={{ current: {} }} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
