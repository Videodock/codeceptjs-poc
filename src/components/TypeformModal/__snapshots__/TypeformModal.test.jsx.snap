// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeformModal /> renders and matches snapshot 1`] = `
<div
  class="ReactTypeformEmbed"
  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; overflow: hidden;"
/>
`;
